android/app/src/main/kotlin/io/vikunja/flutteringvikunja/widget/AppWidget.kt

```diff
@@ -37,36 +47,87 @@ class AppWidget : GlanceAppWidget() {
         }
     }
 
+    // This function cannot be composable otherwise it wont run sometimes when shared prefs isnt changed
+    private fun getTasks(prefs: SharedPreferences) {
+        // These need to be cleared in case this gets run multiple times
+        todayTasks.clear()
+        otherTasks.clear()
+        val gson = Gson()
+        val taskIDChars = prefs.getString("WidgetTaskIDs", null)
 
-    @Composable
-    private fun GlanceContent(context: Context, currentState: HomeWidgetGlanceState) {
-        val prefs = currentState.preferences
-        val tasks: MutableList<String> = ArrayList()
+        var taskIDs: List<String>  = emptyList()
 
-        // First work out how many tasks we gotta pull down
-        val numTasks = prefs.getInt("numTasks", 0)
+        if (taskIDChars != null) {
+            val noBrackets = taskIDChars.substring(1, taskIDChars.length - 1)
+            taskIDs = noBrackets.split(",")
 
-        // Extract all the tasks and put them into that array
-        for (i in 1..numTasks) {
-            val task = prefs.getString(i.toString(), null)
-            task?.let { tasks.add(it) }
+        } else {
+            Log.d("Widget", "There was a problem getting the widget ids")
         }
+        // For some reason if there are no tasks an array will get created with 1 empty/null entry.
+        if (taskIDs.isNotEmpty() && taskIDs[0].isNotEmpty()) {
+            for (taskId in taskIDs) {
+                val taskJSON = prefs.getString(taskId.trim(), null)
+                val task = gson.fromJson(taskJSON, Task::class.java)
+                if (task.today) {
+                    todayTasks.add(task)
+                } else {
+                    otherTasks.add(task)
+                }
+            }
+        }
+    }
+
 
+    private fun doneTask(context: Context, prefs: SharedPreferences, taskID: String) {
+        prefs.edit {
+            putString("completeTask", taskID)
+            commit()
+        }
+        val backgroundIntent = HomeWidgetBackgroundIntent.getBroadcast(
+            context,
+            Uri.parse("appWidget://completeTask")
```

Comment:
Is this a general thing or should this be a custom protocol for the Vikunja app? If the latter, shouldn't this be something like `vikunja-app://completeTask`?

(I don't know much about Android, it's entirely valid if the way you did it is the correct way)